hazelcast/src/main/java/com/hazelcast/spi/impl/merge/SplitBrainUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.merge;
+
+import com.hazelcast.core.HazelcastInstanceAware;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.serialization.SerializationServiceAware;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * Utility class for split-brain healing.
+ *
+ * @since 3.10
+ */
+public final class SplitBrainUtil {
+
+    private SplitBrainUtil() {
+    }
+
+    public static void injectDependencies(Object instance, NodeEngine nodeEngine) {
+        checkNotNull(nodeEngine, "nodeEngine cannot be null");
+        if (instance instanceof HazelcastInstanceAware) {
+            checkNotNull(nodeEngine.getHazelcastInstance(), "HazelcastInstance cannot be null");
+            ((HazelcastInstanceAware) instance).setHazelcastInstance(nodeEngine.getHazelcastInstance());
+        }
+        if (instance instanceof SerializationServiceAware) {
+            checkNotNull(nodeEngine.getSerializationService(), "SerializationService cannot be null");
+            ((SerializationServiceAware) instance).setSerializationService(nodeEngine.getSerializationService());
+        }
```

Comment:
For completeness, you can also invoke: `nodeEngine.getSerializationService().getManagedContext().initialize(instance)`. It will also inject `Node` (in case the instance implements `NodeAware`) and initialise spring beans (if the instance has some spring specific features).